pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -673,8 +673,11 @@ func (r *Replica) applySnapshot(
 	// https://github.com/cockroachdb/cockroach/issues/93251
 	if inSnap.doExcise {
 		exciseSpan := desc.KeySpan().AsRawSpanWithNoLocals()
+		// Since these snapshot SSTs contain RANGEDELs, we can make use of storage
+		// engine optimizations for SSTs that have the RANGEDELs.
+		sstsContainExciseTombstone := true
```

Comment:
We actually need a few other changes to make this work. Currently the SSTs don't contain rangedels. (Also specify that it's not just RangeDel but also RangeKeyDels)

1) We'll need to add a new field to inSnap that specifies if this is a `SharedOrExternalReplicate`. (We could do len(inSnap.sharedSSTs) > 0 but that is slightly wrong). `doExcise` is true for shared snapshots, external snapshots, and local snapshots. We'll be setting this to true here for local snapshots but not for shared/external snapshots.
2) See this. The last bool here specifies if the sstable should skip adding tombstones. We should set that to true for shared and external snapshots but we should flip it to false for local snapshots or the range, so we have the tombstones in the SST:

https://github.com/cockroachdb/cockroach/blob/0b2349225e63c620f66ad57226205709a058106e/pkg/kv/kvserver/store_snapshot.go#L516